src/asus-common.h

```diff
@@ -0,0 +1,246 @@
+#pragma once
+
+// common defines used by libasus (asus.c) and libratbag (driver-asus.c)
+
+#include <linux/input.h>
+#include <stdint.h>
+
+#include "asus-defines.h"
+
+#include "libratbag-private.h"
+#include "libratbag-hidraw.h"
+
+// base request
+
+struct _asus_request {
+	uint16_t cmd;
+	uint8_t params[ASUS_PACKET_SIZE - 2];
+} __attribute__((packed));
+
+union asus_request {
+	struct _asus_request data;
+	uint8_t raw[ASUS_PACKET_SIZE];
+};
+
+// base response
+
+struct _asus_response {
+	uint16_t code;
+	uint8_t results[ASUS_PACKET_SIZE - 2];
+} __attribute__((packed));
+
+union asus_response {
+	struct _asus_response data;
+	uint8_t raw[ASUS_PACKET_SIZE];
+};
+
+// current profile ID and firmware info
+
+struct asus_profile_data {
+	unsigned int profile_id;
+	uint8_t version_primary[3];  // major.minor.build
+	uint8_t version_secondary[3];  // major.minor.build
+} __attribute__((packed));
+
+// button bindings
+
+struct _asus_button {
+	uint8_t action;
+	uint8_t type;
+} __attribute__((packed));
+
+struct _asus_button_data {
+	uint8_t filler0[4];
+	struct _asus_button button[ASUS_MAX_NUM_BUTTON];
+} __attribute__((packed));
+
+union asus_button_data {
+	struct _asus_button_data data;
+	uint8_t raw[ASUS_PACKET_SIZE];
+};
+
+// DPI data
+
+struct _asus_dpi2_data {
+	uint8_t filler0[4];
+	uint16_t dpi[2];  // DPI presets
+	uint16_t rate;  // polling rate
+	uint16_t response;  // button response
+	uint16_t snapping;  // angle snapping (on/off)
+} __attribute__((packed));  // struct for storing 2 DPI presets and extra settings
+
+struct _asus_dpi4_data {
+	uint8_t filler0[4];
+	uint16_t dpi[4];
+	uint16_t rate;
+	uint16_t response;
+	uint16_t snapping;
+} __attribute__((packed));  // struct for storing 4 DPI presets and extra settings
+
+union asus_resolution_data {
+	struct _asus_dpi2_data data2;  // data for 2 DPI presets
+	struct _asus_dpi4_data data4;  // data for 4 DPI presets
+	uint8_t raw[sizeof(struct _asus_dpi4_data)];
+};
+
+// led data
+
+struct _asus_led {
+	uint8_t mode;
+	uint8_t brightness;  // 0-4
+	uint8_t r;
+	uint8_t g;
+	uint8_t b;
+} __attribute__((packed));
+
+struct _asus_led_data {
+	uint8_t filler0[4];
+	struct _asus_led led[ASUS_MAX_NUM_LED];  // LEDs
+} __attribute__((packed));
+
+union asus_led_data {
+	struct _asus_led_data data;
+	uint8_t raw[sizeof(struct _asus_led_data)];
+};
+
+// drv data
+
+struct asus_data {
+	uint8_t is_ready;
+} __attribute__((packed));
```

Comment:
this doesn't need to be `packed` since it's just a software struct. And no need for this to be in the header, this is a driver-only data and can move to `asus-driver.c` (afaict)